### PR TITLE
HOT-116-encryptor-decryptor

### PR DESCRIPTION
**Pull request by sebastian-bojara**

### CODE REVIEW

File: crypton.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+func Encrypt(iv []byte, key []byte, plain string) string {
+	block, err := aes.NewCipher(key)
+	if err != nil {
+		panic(err)
+	}
+
+	plainBytes := []byte(plain)
+	cfb := cipher.NewCFBEncrypter(block, iv)
```
Comment (kamillopata) [inline crypton.go:28]: Didn't we tested different implementations of encryption and they were not compatible?
I thought that we decided to use nodejs lambada to encrypt user private data, because of it's compatibility with php encryption

Comment (kglusinski) [reply crypton.go:28]: Yes, that is true. Only question is if you tested something encrypted by WebAPI and if you were trying to decrypt in WebAPI something encrypt by this code.

I had small differences  (like one or two bytes) when I tried to create this but your implementation is a little bit different, maybe it works.

Comment (sebastian-bojara) [reply crypton.go:28]: yeah, I remember something. But, was it edge cases? Because I tested a few cases and it works.

Comment (kglusinski) [reply crypton.go:28]: It was ~100 random data taken from test database with test keys. Then ~100 random data decrypted by WebAPI. 

But possibly I made some mistake in implementation.

Comment (sebastian-bojara) [reply crypton.go:28]: then I gonna test it again, thanks.

Comment (sebastian-bojara) [reply crypton.go:28]: so, it was wrong, because I was using  aes-256-cfb (cipher.NewCFBEncrypter) while we uses aes-256-ctr (https://github.com/Bidroombv/crypton/blob/master/src/cryptolib.js#L3), some of the characters on the end wasn't decrypted.
I fixed this and tested (also with Crypton lib), and it works <3

**Review by kglusinski — APPROVED** [review]

File: crypton.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+func Encrypt(iv []byte, key []byte, plain string) string {
+	block, err := aes.NewCipher(key)
+	if err != nil {
+		panic(err)
+	}
+
+	plainBytes := []byte(plain)
+	cfb := cipher.NewCTR(block, iv)
```
Comment (kglusinski) [inline crypton.go:28]: actually `ctr` ;)

File: crypton.go [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+)
+
+func encodeBase64(b []byte) string {
+	return base64.StdEncoding.EncodeToString(b)
+}
+
+func decodeBase64(s string) []byte {
+	data, err := base64.StdEncoding.DecodeString(s)
+	if err != nil {
+		panic(err)
```
Comment (kglusinski) [inline crypton.go:16]: wouldn't be better to return error instead of panic? Panic should be used really bad situation. Remember it will crash application.

**Review by kglusinski — APPROVED** [review]

**Review by kamillopata — APPROVED** [review]

**Comment by kamillopata** [comment]
I assumed that you've tried to decrypt some fields from `bidroom_db` with this algo

**Comment by sebastian-bojara** [comment]
> I assumed that you've tried to decrypt some fields from `bidroom_db` with this algo

exactly, even quite old ones.
